lib/sidekiq/manager.rb

```diff
@@ -68,12 +68,19 @@ def stop(deadline)
       sleep PAUSE_TIME
       return if @workers.empty?
 
-      logger.info { "Pausing to allow workers to finish..." }
+      logger.info { "Pausing to allow workers to finish by #{deadline}..." }
       remaining = deadline - ::Process.clock_gettime(::Process::CLOCK_MONOTONIC)
+      num_sleep = 0
       while remaining > PAUSE_TIME
         return if @workers.empty?
         sleep PAUSE_TIME
+        num_sleep += 1
         remaining = deadline - ::Process.clock_gettime(::Process::CLOCK_MONOTONIC)
+
+        # Log out every 30s how much time is left in the deadline
+        if (num_sleep * PAUSE_TIME) % 30 == 0
```

Comment:
Are we setting the options[:timeout] somewhere I'm not seeing?
When I run this locally and on staging, it appears the `:timeout` is 25 (seconds ?)
and so I think we would want to log out sooner than 30s

```ruby
[2] [appboy][development] pry(main)> Sidekiq.options[:timeout]
=> 25
[3] [appboy][development] pry(main)> Sidekiq::DEFAULTS[:timeout]
=> 25
```

Reply:
Yeah it's set via the sidekiq configuration yaml file. The value is set by Chef from here: https://github.com/Appboy/ab_ops/blob/master/cookbooks/cookbooks/platform/templates/default/app/sidekiq/config.yml.erb.

Reply:
It's 600 on most servers and more on w-maint. (I had thought it was 660, 11 minutes, but turns out we set it to 10 minutes). We wouldn't want a value that's <30s because we see plenty of jobs take longer than that.

Reply:
ah, rails console just doesn't load those files so I wasn't seeing the real values

Is there actually a good reason why staging is 18 times larger than production (10800 vs 600)?

```console
$ cat /opt/platform/etc/sidekiq/dashboard_0.yml
---
:require: /opt/platform/current/dashboard
:concurrency: 2
:timeout: 10800
:pidfile: /opt/platform/var/run/sidekiq_dashboard.0.pid

:queues:
...

:index: 0
:tag: dashboard-0
```

Reply:
wow no I'll talk to devops to change this

Reply:
was a staging server: https://github.com/Appboy/ab_ops/blob/d3412088044dbd16c284df3708e81eca0a8784e6/cookbooks/cookbooks/platform/attributes/default.rb#L146